src/runtime/task/mod.rs

```diff
@@ -117,11 +128,11 @@ impl Task {
         let waker = make_waker(id);
         let continuation = Rc::new(RefCell::new(continuation));
 
-        let span = if name == Some("main-thread".to_string()) {
-            parent_span
-        } else {
-            tracing::info_span!(parent: parent_span.id(), "step", i = schedule_len, task = id.0)
-        };
+        let step_span = error_span!(parent: parent_span_id.clone(), "step", task = id.0, i = field::Empty);
+        // Note that this is slightly lazy — we are starting storing at the step_span, but could have gotten the
```

Comment:
Hm, if we know that storing the full stack would be better, why are we cutting corners here?  Why not just do the right thing?

Reply:
Because I don't think it is better / I don't believe this is cutting corners. This comment and the other one of a similar nature are meant as the following: "I really believe the current scheme is correct. However, spans are both weird and tricky. If there ever are any issues, then start by doing the changes in the comments, in order to definitely rule them out.". In other words: I think the current way of doing it is "just right", and that the suggestions are "too much". I'd rather do the "just right" approach, both for overall code cleanness, and for the opportunity to be proven wrong (and thus learn more about Spans), but leave the comments in case I am wrong.